### DISCUSSION TITLE
WebAssembly Text Format (WAT) formatter?

### DISCUSSION THREAD

**Discussion by giffeler** [prettier/prettier#14530] [Ideas]
Surely it should be relatively easy to create a formatter for the Webassembly Text Format ([https://developer.mozilla.org/en-US/docs/WebAssembly/Understanding_the_text_format])? Is anyone already working on it?

**Comment by kachkaev** [comment]
Looks like an opportunity for a [Prettier plugin](https://prettier.io/docs/en/plugins.html). Feel free to create one and we can add it to the [list](https://prettier.io/docs/en/plugins.html#community-plugins).

**Comment by LuluFischettiKY0** [comment]
Hey giffeler!
I can definitely see why a formatter for WebAssembly Text Format would be useful. It would make it so much easier to read and work with! And you're right, it does seem like it would be relatively easy to create.

**Comment by seiyab** [comment]
I've published [prettier-plugin-wat](https://www.npmjs.com/package/prettier-plugin-wat). It's still in alpha and lacks some syntax support. Issues and PRs are welcome.

**Reply by fisker** [reply]
Awesome!

**Reply by giffeler** [reply]
"Gut Ding will Weile haben" (Good things take time). Thank you.
